modules/src/clients/ics07_tendermint/client_state.rs

```diff
@@ -1,30 +1,52 @@
+use crate::core::ics02_client::context::ClientReader;
+use crate::core::ics03_connection::connection::ConnectionEnd;
+use crate::core::ics04_channel::commitment::{AcknowledgementCommitment, PacketCommitment};
+use crate::core::ics04_channel::context::ChannelReaderLightClient;
+use crate::core::ics04_channel::packet::Sequence;
+use crate::core::ics23_commitment::commitment::{
+    CommitmentPrefix, CommitmentProofBytes, CommitmentRoot,
+};
+use crate::core::ics23_commitment::merkle::{apply_prefix, MerkleProof};
+use crate::core::ics24_host::path::{
+    AcksPath, ChannelEndsPath, ClientConsensusStatePath, ClientStatePath, CommitmentsPath,
+    ConnectionsPath, ReceiptsPath, SeqRecvsPath,
+};
+use crate::core::ics24_host::Path;
 use crate::prelude::*;
 
 use core::convert::{TryFrom, TryInto};
 use core::time::Duration;
 
 use ibc_proto::google::protobuf::Any;
 use ibc_proto::ibc::core::client::v1::Height as RawHeight;
-use ibc_proto::ibc::lightclients::tendermint::v1::ClientState as RawClientState;
+use ibc_proto::ibc::core::commitment::v1::MerkleProof as RawMerkleProof;
+use ibc_proto::ibc::lightclients::tendermint::v1::ClientState as RawTmClientState;
 use ibc_proto::protobuf::Protobuf;
+use prost::Message;
 use serde::{Deserialize, Serialize};
 use tendermint_light_client_verifier::options::Options;
+use tendermint_light_client_verifier::types::{TrustedBlockState, UntrustedBlockState};
+use tendermint_light_client_verifier::{ProdVerifier, Verdict, Verifier};
 
+use crate::clients::ics07_tendermint::consensus_state::ConsensusState as TmConsensusState;
 use crate::clients::ics07_tendermint::error::Error;
-use crate::clients::ics07_tendermint::header::Header;
-use crate::core::ics02_client::client_state::UpgradeOptions as CoreUpgradeOptions;
+use crate::clients::ics07_tendermint::header::Header as TmHeader;
+use crate::core::ics02_client::client_state::{
+    ClientState, UpdatedState, UpgradeOptions as CoreUpgradeOptions,
+};
 use crate::core::ics02_client::client_type::ClientType;
-use crate::core::ics02_client::error::Error as Ics02Error;
+use crate::core::ics02_client::consensus_state::ConsensusState;
+use crate::core::ics02_client::error::{Error as Ics02Error, ErrorDetail as Ics02ErrorDetail};
 use crate::core::ics02_client::trust_threshold::TrustThreshold;
 use crate::core::ics23_commitment::specs::ProofSpecs;
-use crate::core::ics24_host::identifier::ChainId;
+use crate::core::ics24_host::identifier::{ChainId, ChannelId, ClientId, ConnectionId, PortId};
 use crate::timestamp::{Timestamp, ZERO_DURATION};
 use crate::Height;
 
 pub const TENDERMINT_CLIENT_STATE_TYPE_URL: &str = "/ibc.lightclients.tendermint.v1.ClientState";
 
-#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub struct ClientState {
+#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
+pub struct TmClientState {
```

Comment:
For the sake of consistency, it would be nice to rename this to `ClientState` (as before), otherwise we could also rename `Header` -> `TmHeader`, etc., but I like the former more as it conforms to [Rust RFC 0356-no-module-prefixes](https://rust-lang.github.io/rfcs/0356-no-module-prefixes.html).

Reply:
[d3251f6](https://github.com/informalsystems/ibc-rs/pull/2570/commits/d3251f6bb647ba8fad25579fe41801c93c6bd466)